src/mongocrypt-kms-ctx.c

```diff
@@ -1443,6 +1521,9 @@ _mongocrypt_kms_ctx_init_kmip_register (mongocrypt_kms_ctx_t *kms_ctx,
                                         uint32_t secretdata_len,
                                         _mongocrypt_log_t *log)
 {
+   BSON_ASSERT_PARAM (kms_ctx);
+   BSON_ASSERT_PARAM (endpoint);
```

Comment:
```suggestion
   BSON_ASSERT_PARAM (kms_ctx);
   BSON_ASSERT_PARAM (endpoint);
   BSON_ASSERT_PARAM (secretdata);
```
Assert `secretdata` is not-null.